tests/Functionnal/AirtableClientTest.php

```diff
@@ -0,0 +1,231 @@
+<?php
+
+namespace Yoanbernabeu\AirtableClientBundle\Tests\Functionnal;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Serializer\Normalizer\ObjectNormalizer;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Yoanbernabeu\AirtableClientBundle\AirtableClient;
+use Yoanbernabeu\AirtableClientBundle\AirtableRecord;
+use Yoanbernabeu\AirtableClientBundle\Tests\Functionnal\Mock\Customer;
+use Yoanbernabeu\AirtableClientBundle\Tests\MockResponse;
+
+class AirtableClientTest extends TestCase
+{
+    private $normalizer;
+
+    protected function setUp(): void
+    {
+        $this->normalizer =  new ObjectNormalizer();
+    }
+
+    /** @test */
+    public function findAll_will_return_airtable_records()
+    {
+        // Setup the dummy HttpClient
+        $httpClient = $this->createHttpClientMock(
```

Comment:
💡 there is a bunch of this except the url, refacto in 1 method private function `request($url)` like you did for `createHttpClientMock`  ResponseInterface and call anywhere you need a mock response

Reply:
But the HttpClientMock allows us to be sure that the call was made isn't it ?